Firestore/core/src/remote/remote_store.cc

```diff
@@ -77,6 +79,15 @@ void RemoteStore::Start() {
   // For now, all setup is handled by `EnableNetwork`. We might expand on this
   // in the future.
   EnableNetwork();
+
+  connectivity_monitor_->AddCallback(
+      [this](ConnectivityMonitor::NetworkStatus /*ignored*/) {
```

Comment:
Is it necessary/desirable to run this logic when the network status is "unavailable"?

Reply:
Probably not - but note that https://github.com/firebase/firebase-ios-sdk/blob/8633f5f6a3ca936085c9ee90680b45660eb19dd0/Firestore/core/src/remote/grpc_connection.cc#L296 kills the GRPC channel no matter what.